Services/CustomDeviceModels.cs

```diff
@@ -86,7 +91,9 @@ public async Task<IEnumerable<DeviceModel>> GetListAsync()
             }
             catch (Exception e)
             {
-                this.log.Error("Unable to parse device models loaded from storage", e);
+                var msg = "Unable to parse device models loaded from storage";
+                this.log.Error(msg, e);
+                this.diagnosticsLogger.LogServiceErrorAsync(msg, e);
                 throw new ExternalDependencyException("Unable to parse device models loaded from storage", e);
```

Comment:
use msg instead. Same applies below where you are logging and throwing exception